lib/swagger-client.js

```diff
@@ -1464,18 +1464,7 @@ Operation.prototype.encodeQueryParam = function(arg) {
  * TODO revisit, might not want to leave '/'
  **/
 Operation.prototype.encodePathParam = function(pathParam) {
-  var encParts, part, parts, i, len;
```

Comment:
This can be done with a much simpler approach:

```js
return pathParam.toString().split('/').map(encodeURIComponent).join('/')
```

Please update the code and and add test if possible.